components/brave_wallet/browser/wallet_data_files_installer.cc

```diff
@@ -0,0 +1,187 @@
+    0xc0, 0x3e, 0x9b, 0x2a, 0xc2, 0xf5, 0x6a, 0x37, 0x71, 0x67};
+const char kWalletDataFilesDisplayName[] = "Brave Wallet data files";
+const char kWalletBaseDirectory[] = "BraveWallet";
+
+static_assert(base::size(kWalletDataFilesSha2Hash) == crypto::kSHA256Length,
+              "Wrong hash length");
+
+}  // namespace
+
+class WalletDataFilesInstallerPolicy
+    : public component_updater::ComponentInstallerPolicy {
+ public:
+  WalletDataFilesInstallerPolicy();
+  ~WalletDataFilesInstallerPolicy() override = default;
+
+ private:
+  // The following methods override ComponentInstallerPolicy.
+  bool SupportsGroupPolicyEnabledComponentUpdates() const override;
+  bool RequiresNetworkEncryption() const override;
+  update_client::CrxInstaller::Result OnCustomInstall(
+      const base::DictionaryValue& manifest,
+      const base::FilePath& install_dir) override;
+  void OnCustomUninstall() override;
+  bool VerifyInstallation(const base::DictionaryValue& manifest,
+                          const base::FilePath& install_dir) const override;
+  void ComponentReady(const base::Version& version,
+                      const base::FilePath& path,
+                      std::unique_ptr<base::DictionaryValue> manifest) override;
+  base::FilePath GetRelativeInstallDir() const override;
+  void GetHash(std::vector<uint8_t>* hash) const override;
+  std::string GetName() const override;
+  update_client::InstallerAttributes GetInstallerAttributes() const override;
+
+  std::vector<mojom::ERCTokenPtr> TokenListReady(
+      const base::FilePath& install_dir,
+      std::unique_ptr<base::DictionaryValue> manifest);
+  void UpdateTokenRegistry(std::vector<mojom::ERCTokenPtr>);
+
+  WalletDataFilesInstallerPolicy(const WalletDataFilesInstallerPolicy&) =
+      delete;
+  WalletDataFilesInstallerPolicy operator=(
+      const WalletDataFilesInstallerPolicy&) = delete;
+};
+
+WalletDataFilesInstallerPolicy::WalletDataFilesInstallerPolicy() = default;
+
+bool WalletDataFilesInstallerPolicy::
+    SupportsGroupPolicyEnabledComponentUpdates() const {
+  return false;
+}
+
+bool WalletDataFilesInstallerPolicy::RequiresNetworkEncryption() const {
+  return false;
+}
+
+update_client::CrxInstaller::Result
+WalletDataFilesInstallerPolicy::OnCustomInstall(
+    const base::DictionaryValue& manifest,
+    const base::FilePath& install_dir) {
+  return update_client::CrxInstaller::Result(update_client::InstallError::NONE);
+}
+
+void WalletDataFilesInstallerPolicy::OnCustomUninstall() {}
+
+void WalletDataFilesInstallerPolicy::ComponentReady(
+    const base::Version& version,
+    const base::FilePath& path,
+    std::unique_ptr<base::DictionaryValue> manifest) {
+  base::ThreadPool::PostTaskAndReplyWithResult(
+      FROM_HERE, {base::MayBlock(), base::TaskPriority::BEST_EFFORT},
+      base::BindOnce(&WalletDataFilesInstallerPolicy::TokenListReady,
+                     base::Unretained(this), path, std::move(manifest)),
```

Comment:
this is not safe. I know widevine also has base::Unretained, but that isn't safe either. At least for widevine it's USER_VISIBLE so it's higher priority and less likely to get called after the installer policy is destroyed

Reply:
Posted this issue to track it: https://github.com/brave/brave-browser/issues/17509